tests/Unit/NumericalAlgorithms/Spectral/SwshTestHelpers.hpp

```diff
@@ -40,6 +51,116 @@ std::complex<double> spin_weighted_spherical_harmonic(int s, int l, int m,
 template <typename DerivativeKind>
 std::complex<double> derivative_of_spin_weighted_spherical_harmonic(
     int s, int l, int m, double theta, double phi) noexcept;
+
+// Performs the generation of spin-weighted coefficients into a supplied
+// ComplexModalVector `to_fill`, passed by pointer. The resulting random
+// coefficients must then be adjusted in two ways:
+// - The modes l < |s| are always zero, so must have zero coefficient to be
+//   compatible with tests
+// - Modes with m=0 obey reality conditions, so must also be adjusted
```

Comment:
is there documentation, e.g. for TransformJob, that discusses these conditions that could be pointed to?

Reply:
No, but at current the coefficients are not considered user-facing, so I've omitted a great deal of discussion of the rather unusual way that libsharp has chosen to do it.

Reply:
Alright, I assume the unusual ways are part of the libsharp documentation. If not, it would be good to add it somewhere, just so whoever has to interact with it in 10 years from now isn't completely helpless :P

Reply:
That's a pretty strong assumption. I'll try to give an outline in code comments in the Coefficients file.

Reply:
haha fair. Okay, I think it would be good to add comments on all the things you struggled with while getting this to work. That should help the next person go through less pain :)